src/raps/actions/ens.ts

```diff
@@ -0,0 +1,440 @@
+  let maxPriorityFeePerGas;
+  try {
+    maxFeePerGas = selectedGasFee?.gasFeeParams?.maxFeePerGas?.amount;
+    maxPriorityFeePerGas =
+      selectedGasFee?.gasFeeParams?.maxPriorityFeePerGas?.amount;
+
+    logger.sentry(`[${actionName}] about to ${type}`, {
+      ...parameters,
+    });
+    const nonce = baseNonce ? baseNonce + index : null;
+
+    switch (type) {
+      case ENSRegistrationTransactionType.COMMIT:
+        tx = await executeCommit(
+          name,
+          duration,
+          ownerAddress,
+          rentPrice,
+          salt,
+          gasLimit,
+          maxFeePerGas,
+          maxPriorityFeePerGas,
+          wallet,
+          nonce
+        );
+        break;
+      case ENSRegistrationTransactionType.REGISTER_WITH_CONFIG:
+        tx = await executeRegisterWithConfig(
+          name,
+          duration,
+          ownerAddress,
+          rentPrice,
+          salt,
+          gasLimit,
+          maxFeePerGas,
+          maxPriorityFeePerGas,
+          wallet,
+          nonce
+        );
+        break;
+      case ENSRegistrationTransactionType.SET_TEXT:
+        tx = await executeSetText(
+          name,
+          recordKey,
+          recordValue,
+          gasLimit,
+          maxFeePerGas,
+          maxPriorityFeePerGas,
+          wallet,
+          nonce
+        );
+        break;
+      case ENSRegistrationTransactionType.MULTICALL:
+        tx = await executeMulticall(
+          name,
+          records,
+          gasLimit,
+          maxFeePerGas,
+          maxPriorityFeePerGas,
+          wallet,
+          nonce
+        );
+        break;
+      case ENSRegistrationTransactionType.SET_NAME:
+        tx = await executeSetName(
+          name,
+          ownerAddress,
+          gasLimit,
+          maxFeePerGas,
+          maxPriorityFeePerGas,
+          wallet,
+          nonce
+        );
+    }
+  } catch (e) {
+    logger.sentry(`[${actionName}] Error executing`);
+    captureException(e);
+    throw e;
+  }
+
+  logger.log(`[${actionName}] response`, tx);
+
+  const nativeAsset = await ethereumUtils.getNetworkNativeAsset(
+    NetworkTypes.mainnet
+  );
+  const newTransaction = {
+    amount: 0,
+    asset: nativeAsset,
+    data: tx.data,
+    from: ownerAddress,
+    gasLimit,
+    hash: tx?.hash,
+    maxFeePerGas,
+    maxPriorityFeePerGas,
+    nonce: tx?.nonce,
+    to: tx?.to,
+    value: toHex(tx.value),
+  };
+  logger.log(`[${actionName}] adding new txn`, newTransaction);
+  // @ts-expect-error Since src/redux/data.js is not typed yet, `accountAddress`
+  // being a string conflicts with the inferred type of "null" for the second
+  // parameter.
+  await dispatch(dataAddNewTransaction(newTransaction, ownerAddress, true));
+  return tx?.nonce;
+};
+
+const commitENS = async (
+  wallet: Wallet,
+  currentRap: Rap,
+  index: number,
+  parameters: RapActionParameters,
+  baseNonce?: number
+): Promise<number | undefined> => {
+  const actionName = 'commitENS';
```

Comment:
Can u use RapActionType.commitENS instead?  same for the other fns

Reply:
https://github.com/rainbow-me/rainbow/pull/2975